Common/Util/ParallelRunner.cs

```diff
@@ -152,19 +152,16 @@ public void Dispose()
         {
             lock (_sync)
             {
-                if (_holdQueue != null) _holdQueue.Dispose();
-                if (_processQueue != null) _processQueue.Dispose();
-
-                // Wait for _holdQueue disposal be completed
-                Thread.Sleep(10000);
-
                 if (_processQueueThread != null && _processQueueThread.IsAlive) _processQueueThread.Abort();
```

Comment:
Love the sleep removal 👍
 
While we're touching this `Dispose` method, we could also try to terminate more gracefully, perhaps replacing the `Thread.Abort` with `Thread.Join` or something else?

Reply:
Good idea, I'll add a combination of both (`Join` with timeout) 👍

Reply:
Great ideas guys, love it!